packages/api/src/auth/decoders/auth0.ts

```diff
@@ -16,37 +56,25 @@ import jwksClient from 'jwks-rsa'
  * ```
  *
  * You can use `sub` as a stable reference to your user, but  if you want the email
- * addres you can set a context object[^0] in rules[^1]:
+ * address you can set a context object[^0] in rules[^1]:
  *
  * ^0: https://auth0.com/docs/rules/references/context-object
  * ^1: https://manage.auth0.com/#/rules/new
  *
  */
-export const verifyAuth0Token = (
+
+export const verifyAuth0Token = async (
   bearerToken: string
 ): Promise<null | Record<string, unknown>> => {
   return new Promise((resolve, reject) => {
-    const { AUTH0_DOMAIN, AUTH0_AUDIENCE } = process.env
-    if (!AUTH0_DOMAIN || !AUTH0_AUDIENCE) {
-      throw new Error(
-        '`AUTH0_DOMAIN` or `AUTH0_AUDIENCE` env vars are not set.'
-      )
-    }
-
-    const client = jwksClient({
-      jwksUri: `https://${AUTH0_DOMAIN}/.well-known/jwks.json`,
-    })
-
     jwt.verify(
       bearerToken,
-      (header, callback) => {
-        client.getSigningKey(header.kid as string, (error, key) => {
-          callback(error, key.getPublicKey())
-        })
+      async (header) => {
+        return getAuth0PublicKey(header)
```

Comment:
Sorry I left a note for myself earlier that I meant to expand on but accidentally submitted in note form (i.e. incomplete sentences). What I meant to say was that I think we may still need to use `callback` here. Here's the types of `jwt.verify`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f6624204b599ef6d23a094b428edc8395d04e78d/types/jsonwebtoken/index.d.ts#L147-L155; emphasis on the `void` return.

I couldn't get this PR to build yet (the `yarn.lock` has a ton of changes for just adding a devDependency, `nock`), so I copied all the changes to a new local branch that I just checked-out from main so I could try them out on a local auth0 project of mine via `yarn rwfw project:sync`.

Right now it seems the whole app is stuck on `Processing GraphQL Parameters`

<img width="1523" alt="stuck-on-processing-graphql-params" src="https://user-images.githubusercontent.com/32992335/169525113-8807a2ea-c351-4351-8c37-0552e0b20c46.png">

The value of the `loading` property from `useAuth` is also `true` and doesn't change, so it looks like something's not completing here and I'm assuming that it's cause we're not calling `callback` anymore. I'll test my theory and look at the tests you added

Reply:
Here's an issue about `jwt.verify`'s API that's been open for quite some time and is still pretty active: https://github.com/auth0/node-jsonwebtoken/issues/111#issuecomment-904557399. @pi0neerpat I saw you in that thread (small world)—what's the story there?


Also in that thread, Joël Galeran from Prisma threw out [jose](https://github.com/panva/jose) as an alternative: https://github.com/auth0/node-jsonwebtoken/issues/111#issuecomment-793944893.